gateway/src/main/java/org/sample/payment/gateway/controller/PaymentController.java

```diff
@@ -0,0 +1,61 @@
+package org.sample.payment.gateway.controller;
+
+import org.sample.payment.gateway.exception.CustomValidationException;
+import org.sample.payment.gateway.model.*;
+import org.sample.payment.gateway.service.PaymentRecordService;
+import org.sample.payment.gateway.service.PaymentDTO;
+import org.sample.payment.gateway.service.ValidationImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static org.sample.payment.gateway.logger.LoggerHelper.LOGGER;
+
+
+@RestController
+public class PaymentController {
+
+    public PaymentRecordService paymentRecordService;
+    public PaymentDTO paymentDTO;
+
+    public ValidationImpl validation;
+
+    @Autowired
+    public PaymentController(PaymentRecordService paymentRecordService, PaymentDTO paymentToPaymentRecord, ValidationImpl validation) {
+        this.paymentRecordService = paymentRecordService;
+        this.paymentDTO = paymentToPaymentRecord;
+        this.validation = validation;
+    }
+
+    @GetMapping(value = "/payment/get", produces = "application/json")
+    public PaymentResponse getPayment(@RequestBody String invoice){
```

Comment:
Can we receive request body as input method parameter instead of doing unmarshaling separately?